src/coreclr/inc/corinfo.h

```diff
@@ -689,34 +690,29 @@ enum CorInfoCallConv
     CORINFO_CALLCONV_PARAMTYPE  = 0x80,     // Passed last. Same as CORINFO_GENERICS_CTXT_FROM_PARAMTYPEARG
 };
 
-#ifdef UNIX_X86_ABI
-inline bool IsCallerPop(CorInfoCallConv callConv)
-{
-    unsigned int umask = CORINFO_CALLCONV_STDCALL
-                       | CORINFO_CALLCONV_THISCALL
-                       | CORINFO_CALLCONV_FASTCALL;
-
-    return !(callConv & umask);
-}
-#endif // UNIX_X86_ABI
-
 // Represents the calling conventions supported with the extensible calling convention syntax
 // as well as the original metadata-encoded calling conventions.
-enum CorInfoUnmanagedCallConv
-{
-    // These correspond to CorUnmanagedCallingConvention
-    CORINFO_UNMANAGED_CALLCONV_UNKNOWN,
-    CORINFO_UNMANAGED_CALLCONV_C,
-    CORINFO_UNMANAGED_CALLCONV_STDCALL,
-    CORINFO_UNMANAGED_CALLCONV_THISCALL,
-    CORINFO_UNMANAGED_CALLCONV_FASTCALL
+enum class CorInfoCallConvExtension
+{
+    Managed,
+    C,
+    Stdcall,
+    Thiscall,
+    Fastcall
     // New calling conventions supported with the extensible calling convention encoding go here.
 };
 
+#ifdef UNIX_X86_ABI
+inline bool IsCallerPop(CorInfoCallConvExtension callConv)
+{
+    return callConv == CorInfoCallConvExtension::Managed || callConv == CorInfoCallConvExtension::C;
+}
+#endif // UNIX_X86_ABI
+
 // Determines whether or not this calling convention is an instance method calling convention.
-inline bool callConvIsInstanceMethodCallConv(CorInfoUnmanagedCallConv callConv)
+inline bool callConvIsInstanceMethodCallConv(CorInfoCallConvExtension callConv)
```

Comment:
Nit: I think helper methods like this belong to the JIT, not to the JIT/EE interface.

Reply:
I agree, but I'm not sure where to put them otherwise.

Reply:
`coreclr/jit/target.h` ?

Reply:
I can try that. target.h doesn't include corinfo.h, so I'm concerned about introducing an implicit dependency there.

Reply:
I'll put it in jit.h since that seems to be how corinfo.h gets included in the jit.

Reply:
That doesn't work since gentree.h doesn't include jit.h.

I'm just going to leave it as is for now.